File: silver/2018.cpp
#include <iostream>
using namespace std;

int main()
{
	int n;
	cin >> n;

	int start = 1, end = 1, cnt = 0, sum = 1;
	while(start <= n && end <= n) {
		if (sum == n) {
			cnt++;
			end++;
			sum += end;
		}
		else if (sum > n){
			sum -= start;
			start++;
		}
		else {
			end++;
			sum += end;
		}
	}
	cout << cnt << endl;
}

File: silver/2018.py
import sys
input = sys.stdin.readline

n = int(input())

cnt = 1
start = 1
end = 1
tmp = 0
while start < n/2:
    tmp += end
    if tmp > n:
        start += 1
        end = start
        tmp = 0
    elif tmp == n:
        cnt += 1
        start += 1
        end = start
        tmp = 0
    else:
        end += 1
print(cnt)
